psi4/driver/procrouting/proc.py

```diff
@@ -2209,6 +2209,10 @@ def run_scf(name, **kwargs):
     if "dft_functional" in kwargs:
         dft_func = True
 
+    # See if we're doing TDSCF after, keep JK if so 
+    if sum(core.get_global_option("TDSCF_STATES_PER_IRREP")) > 0:
+        core.set_global_option("SAVE_JK", True)
```

Comment:
need to add to the `OptionsState` above via `['SCF', 'SAVE_JK']`. this resets the option when the run_scf exits, so that options look like how user set them, not with interference from driver. any reason not to do `get_option("SCF", "TDSCF_STATES_PER_IRREP")` and `set_local_option("SAVE_JK")`? probably as-is your routine wouldn't work if input was `set scf tdscf_states_per_irrep [1]`.